spec/system/admin/products_v3/actions_spec.rb

```diff
@@ -34,42 +34,59 @@
   describe "column selector" do
     let!(:product) { create(:simple_product) }
 
-    before do
-      visit admin_products_url
-    end
+    context "with one producer only" do
+      before do
+        visit admin_products_url
+      end
+
+      it "hides column and remembers saved preference" do
+        # Name shows by default
+        expect(page).to have_checked_field "Name"
+        expect(page).to have_selector "th", text: "Name"
+        expect_other_columns_visible
 
-    it "hides column and remembers saved preference" do
-      # Name shows by default
-      expect(page).to have_checked_field "Name"
-      expect(page).to have_selector "th", text: "Name"
-      expect_other_columns_visible
+        # Producer is hidden by if only one producer is present
+        expect(page).not_to have_checked_field "Producer"
+        expect(page).not_to have_selector "th", text: "Producer"
 
-      # Name is hidden
-      ofn_drop_down("Columns").click
-      within ofn_drop_down("Columns") do
-        uncheck "Name"
+        # Name is hidden
+        ofn_drop_down("Columns").click
+        within ofn_drop_down("Columns") do
+          uncheck "Name"
+        end
+        expect(page).not_to have_selector "th", text: "Name"
+        expect_other_columns_visible
+
+        # Preference saved
+        click_on "Save as default"
+        expect(page).to have_content "Column preferences saved"
+        refresh
+
+        # Preference remembered
+        ofn_drop_down("Columns").click
+        within ofn_drop_down("Columns") do
+          expect(page).to have_unchecked_field "Name"
+        end
+        expect(page).not_to have_selector "th", text: "Name"
+        expect_other_columns_visible
```

Comment:
Shouldn't we check if the producer column can be made visible if the user has updated his column preferences ?

Reply:
I thought if the producer column option is displayed as unchecked under the column selector, then checking it should display the column as per the existing behavior.
But you are right, it's better to be safe so I've added the spec here; https://github.com/openfoodfoundation/openfoodnetwork/pull/12854/commits/243a4a55b435a4afef9383d17c4afc9a2e8e1b22